### PR TITLE
Refactor agent bootstrap

### LINKED ISSUE #23049: refactor the agent packaging
## Preface
currently the agent module use `maven shade plugin` as the solution for dependency conflict, however this makes a lot of dependency duplication in different plugin modules, followings are the plugins :

image

the shaded guava are duplicated in all plugin artifact:
image
image

## Missions
to refactor the agent packaging, we need to refactor followings :
- [x] add shaded dependency in bootstrap module
- [x] add exclusions in plugins to make the plugin artifact smaller
- [x] extract the metrics-core seperately

### Shaded Dependencies
image

### Reduce Size
**before**
image

**after**
image

 the plugins reduced from **63.2MB** to **12.1MB**

### extract metrics-core
image

### Prometheus
image

### Zipkin
image

### PR DESCRIPTION
**Pull request by taojintianxia**
Fixes #23049, #23118

Changes proposed in this pull request:
  - add shaded dependency in bootstrap module
  - add exclusions in plugins to make the plugin artifact smaller
  - extract the metrics-core seperately

**current structure of agent distribution:**
image
image
image

**tested on localhost:**
prometheus:
image

zipkin:
image

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: agent/bootstrap/src/main/java/org/apache/shardingsphere/agent/bootstrap/plugin/loader/AgentPluginLoader.java [hunk 1]
```diff
@@ -46,8 +48,15 @@ public final class AgentPluginLoader {
      * @throws IOException IO exception
      */
     public static Collection<PluginJar> load() throws IOException {
-        File[] jarFiles = AgentPathBuilder.getPluginPath().listFiles(each -> each.getName().endsWith(".jar"));
-        if (null == jarFiles) {
+        List<File> jarFiles = new LinkedList<>();
+        AgentPathBuilder.getPluginClassPaths().forEach(
```
Comment (terrymanu) [inline agent/bootstrap/src/main/java/org/apache/shardingsphere/agent/bootstrap/plugin/loader/AgentPluginLoader.java:52]: extract one method

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: agent/bootstrap/src/main/java/org/apache/shardingsphere/agent/bootstrap/path/AgentPathBuilder.java [hunk 2]
```diff
@@ -71,7 +74,8 @@ private static File getFileInResource(final String url, final String classResour
         return new File(classLocation);
     }
     
-    private static File buildAgentPluginPath() {
-        return new File(String.join("/", agentPath.getPath(), "plugins"));
+    private static List<File> buildAgentPluginPath() {
```
[the remark this answers is not in the document]
Comment (taojintianxia) [reply agent/bootstrap/src/main/java/org/apache/shardingsphere/agent/bootstrap/path/AgentPathBuilder.java:77]: resolved

[the remark this answers is not in the document]
Comment (taojintianxia) [reply agent/bootstrap/src/main/java/org/apache/shardingsphere/agent/bootstrap/path/AgentPathBuilder.java:77]: > Rename to buildAgentPluginPaths

resolved

File: agent/bootstrap/src/main/java/org/apache/shardingsphere/agent/bootstrap/plugin/loader/AgentPluginLoader.java (hunk 1 above)
Comment (taojintianxia) [reply agent/bootstrap/src/main/java/org/apache/shardingsphere/agent/bootstrap/plugin/loader/AgentPluginLoader.java:52]: > extract one method

resolved

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -219,4 +219,5 @@ jobs:
       - name: Run Integration Test
         run: |
           ./mvnw -B clean install -f test/e2e/agent/plugins/metrics/pom.xml -Dspotless.apply.skip=true -Pit.env.metrics
+          ./mvnw clean install -B -T1C -Dmaven.javadoc.skip=true -Djacoco.skip=true -Drat.skip=true -DskipITs -DskipTests -Dcheckstyle.skip=true -Prelease
```
[the remark this answers is not in the document]
Comment (taojintianxia) [reply .github/workflows/e2e.yml:222]: > Check why metrics do not need -Prelease

resolved

File: agent/bootstrap/pom.xml [hunk 4]
```diff
@@ -47,9 +52,8 @@
             <scope>compile</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.shardingsphere</groupId>
-            <artifactId>shardingsphere-agent-plugin-core</artifactId>
-            <version>${project.version}</version>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-transport</artifactId>
         </dependency>
```
[the remark this answers is not in the document]
Comment (taojintianxia) [reply agent/bootstrap/pom.xml:57]: > Bootstrap do not need netty

need further communication. this time, there are something contradictory.

[the remark this answers is not in the document]
Comment (taojintianxia) [reply agent/bootstrap/pom.xml:58]: already removed from bootstrap

**Review by terrymanu — APPROVED** [review]

**Pull request merged**